core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp

```diff
@@ -791,24 +828,60 @@ struct ParallelReduceSpecialize<FunctorType, TeamPolicyInternal<PolicyArgs...>,
                            ? OpenMPTargetExec::MAX_ACTIVE_TEAMS
                            : league_size;
 
+    const size_t shmem_size_L0 = p.scratch_size(0, team_size);
+    const size_t shmem_size_L1 = p.scratch_size(1, team_size);
     OpenMPTargetExec::resize_scratch(PolicyType::member_type::TEAM_REDUCE_SIZE,
-                                     0, 0);
+                                     shmem_size_L0, shmem_size_L1);
     void* scratch_ptr = OpenMPTargetExec::get_scratch_ptr();
 
     ValueType result = ValueType();
 
+    // Maximum active teams possible.
+    int max_active_teams = OpenMPTargetExec::MAX_ACTIVE_THREADS / team_size;
+
+    int* lock_array = OpenMPTargetExec::get_lock_array(max_active_teams);
+
+#pragma omp target teams distribute parallel for is_device_ptr(lock_array)
+    for (int i = 0; i < max_active_teams; ++i) {
+      lock_array[i] = 0;
+    }
```

Comment:
why do this again and again (i probably have overlooked that in the ParallelFor too) I mean we only need to do it when reallocating or (so the get_lock_array could do that). Since otherwise this thing should be back to zero anyway (a kernel will never leave it not being zero unless it crashes).

Reply:
Yes, I was initializing it in case a previous kernel crashes.